includes/class-wc-dintero-hp-checkout.php

```diff
@@ -2789,134 +2729,159 @@ public function process_refund( $order_id, $amount = null, $reason = '' ) {
 
-			$order_total_amount = absint( strval( floatval( $order->get_total() ) * 100 ) );
+		if ( empty( $transaction ) ) {
+			$transaction = $this->get_transaction( $transaction_id );
+		}
 
-			if ( array_key_exists( 'status', $transaction ) &&
-				 array_key_exists( 'amount', $transaction ) &&
-				 'AUTHORIZED' === $transaction['status'] &&
-				 $transaction['amount'] >= $order_total_amount ) {
-				$access_token = $this->get_access_token();
-				$api_endpoint = $this->checkout_endpoint . '/transactions';
+		$order_total_amount = absint( strval( floatval( $order->get_total() ) * 100 ) );
 
-				$items = array();
-
-				$counter = 0;
-				foreach ( $order->get_items() as $order_item ) {
-					$counter ++;
-					$line_id                = strval( $counter );
-					$item_total_amount      = absint( strval( floatval( $order_item->get_total() ) * 100 ) );
-					$item_tax_amount        = absint( strval( floatval( $order_item->get_total_tax() ) * 100 ) );
-					$item_line_total_amount = absint( strval( floatval( $order->get_line_total( $order_item,
-							true ) ) * 100 ) );
-					$item_tax_percentage    = $item_total_amount ? ( round( ( $item_tax_amount / $item_total_amount ),
-							2 ) * 100 ) : 0;
-					$item                   = array(
-						'id'          => 'item_' . $counter,
-						'description' => $order_item->get_name(),
-						'quantity'    => $order_item->get_quantity(),
-						'vat_amount'  => $item_tax_amount,
-						'vat'         => $item_tax_percentage,
-						'amount'      => $item_line_total_amount,
-						'line_id'     => $line_id
-					);
-					array_push( $items, $item );
-				}
+		if (isset($transaction['error'])) {
+			$order->add_order_note(__('Could not capture transaction: ' . $transaction['message']));
+			$order->save_meta_data();
+			return false;
+		}
 
-				if ( count( $order->get_shipping_methods() ) > 0 ) {
-					$counter ++;
-					$line_id                = strval( $counter );
-					$item_total_amount      = absint( strval( floatval( $order->get_shipping_total() ) * 100 ) );
-					$item_tax_amount        = absint( strval( floatval( $order->get_shipping_tax() ) * 100 ) );
-					$item_line_total_amount = $item_total_amount + $item_tax_amount;
-					$item_tax_percentage    = $item_total_amount ? ( round( ( $item_tax_amount / $item_total_amount ),
-							2 ) * 100 ) : 0;
-
-					$item = array(
-						'id'          => 'shipping',
-						'description' => 'Shipping: ' . $order->get_shipping_method(),
-						'quantity'    => 1,
-						'vat_amount'  => $item_tax_amount,
-						'vat'         => $item_tax_percentage,
-						'amount'      => $item_line_total_amount,
-						'line_id'     => $line_id
-					);
-					array_push( $items, $item );
-				}
+		if ($this->extract('status', $transaction) !== 'AUTHORIZED'
+			|| $this->extract('amount', $transaction) != $order_total_amount
+		) {
+			$order->add_order_note(__(
+				'Could not capture transaction: Transaction status is wrong or order and transaction amounts do not match.'
```

Comment:
Would help if the note mentioned the transaction status and the amounts of both order and transaction

Reply:
@andersem , ok, will correct that.